tasks/managed/push-snapshot/push-snapshot.yaml

```
@@ -183,24 +200,61 @@ spec:
             fi
             # Push the source image with the source tag here. The source image will be
             # pushed with the provided tags below in the loop
+            wait_for_slot
             push_image "${source_container_digest}" "${name}" "${sourceContainer}" \
-              "${repository}" "${source_tag}" ""
+              "${repository}" "${source_tag}" "" > "$TMP_RESULTS_DIR/${name}-${source_tag}.out" 2>&1 &
+            jobs+=($!)
+            ((++REQUEST_COUNT))
+            echo "Request Count: $REQUEST_COUNT"
           fi
 
           for tag in $(jq -r '.[]' <<< "$imageTags") ; do
+            wait_for_slot
             # Push the container image
             push_image "${origin_digest}" "${name}" "${containerImage}" "${repository}" "${tag}" \
-            "$platform"
+            "$platform" > "$TMP_RESULTS_DIR/${name}-${tag}.out" 2>&1 &
+            jobs+=($!)
+            ((++REQUEST_COUNT))
+            echo "Request Count: $REQUEST_COUNT"
 
             # This variable will only exist if the above logic determined the source container should
             # be pushed for this component
             if [ -n "${source_container_digest-}" ] ; then
+              wait_for_slot
               push_image "${source_container_digest}" "${name}" "${sourceContainer}" \
-                "${repository}" "${tag}-source" ""
+                "${repository}" "${tag}-source" "" > "$TMP_RESULTS_DIR/${name}-${tag}-source.out" 2>&1 &
+              jobs+=($!)
+              ((++REQUEST_COUNT))
+              echo "Request Count: $REQUEST_COUNT"
             fi
           done
         done
-        echo -n "${RESULTS_JSON}" | tee "$RESULTS_FILE"
+
+        success=true
+        echo "Waiting for all jobs to finish...."
+        for job in "${jobs[@]}"; do
+            if ! wait "$job"; then
```

But what about all the waits that happen as part of `wait_for_slot`? There is no handling of the failures there, so those would still fail immediately, no?

---

Yeah, you're correct. The `wait_for_slot `would fail. Sorry for all this. We could use `wait -n || true`, which would mask any errors and then catch them at the end. .... that's what I am thinking anyway.

---

You should be able to use the same mechanism there - by default, variables used in functions have global scope. So you can access the same variable. So you should set `success=true` before the first use of `wait` (via the function or directly) and then in the function you can also have the same kind of condition you have here.

---

should be fixed now